In [53]:
import numpy as np
import pandas as pd
df = pd.read_csv('data/hotels_merged.csv')
df=df.drop(columns = 'Unnamed: 0')

In [54]:
df.sort_values("hotelName", inplace = True) 

In [55]:
df.drop_duplicates(subset ="hotelName", 
                     keep = 'first', inplace = True)

In [56]:
df.drop(df.columns[[10,11,12,13,14,15,16,17]],axis =1, inplace = True)

In [57]:
df.drop(df[df['priceRange']=='none'].index, axis =0, inplace = True)

df['priceRange'].unique()

df['priceRange'].replace({'$':'1', '$$':'2','$$$':'3','$$$$':'4'}
                          , inplace = True)

df['priceRange'] = pd.to_numeric(df['priceRange'])

In [58]:
df

hotelName  \
35     AC Hotel by Marriott National Harbor Washingto...   
66                                   AKA Virginia Square   
101                                      AKA White House   
129                                     Alexandria Motel   
138                               Alexandria-Towne Motel   
...                                                  ...   
11369                             Washington Plaza Hotel   
11411         West End part of the Georgetown Collection   
11448          WhyHotel Arlington Pentagon Columbia Pike   
11493                           William F. Bolger Center   
11524                                 Windsor Park Hotel   

                                                     url  zipcode  \
35     https://www.yelp.com/biz/ac-hotel-by-marriott-...    20745   
66     https://www.yelp.com/biz/aka-virginia-square-a...    22201   
101    https://www.yelp.com/biz/aka-white-house-washi...    20006   
129    https://www.yelp.com/biz/alexandria-motel-alex...    22306   
138    https://www.yelp.com/biz/alexandria-towne-mote...    22314   
...                                                  ...      ...   
11369  https://www.yelp.com/biz/washington-plaza-hote...    20005   
11411  https://www.yelp.com/biz/west-end-part-of-the-...    20037   
11448  https://www.yelp.com/biz/whyhotel-arlington-pe...    22204   
11493  https://www.yelp.com/biz/william-f-bolger-cent...    20854   
11524  https://www.yelp.com/biz/windsor-park-hotel-wa...    20008   

       totalRatings  hotelRate  totalPhotos  priceRange        highlights_1  \
35               81        3.5          174           2       capital wheel   
66                6        3.5           11           2                none   
101              61        3.5           97           3         one bedroom   
129               3        4.5            4           1                none   
138              11        4.5           11           1                  DC   
...             ...        ...          ...         ...                 ...   
11369           229        3.5          170           2       Thomas Circle   
11411            46        3.0           49           2               Metro   
11448             3        0.0           31           2                none   
11493            68        3.5          143           2  Stained Glass Hall   
11524            30        3.0           63           2         mini fridge   

      highlights_2 highlights_3  
35         Gaylord   few nights  
66            none         none  
101        Farmers    hotel bar  
129           none         none  
138         Keurig         Neil  
...            ...          ...  
11369  white house    sized bed  
11411   sized beds      Western  
11448         none         none  
11493        Sarah      convent  
11524  actual keys        cable  

[241 rows x 10 columns]

In [59]:
df[df=='none'].count()

/opt/anaconda3/envs/python38/lib/python3.8/site-packages/pandas/core/ops/array_ops.py:253: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  res_values = method(rvalues)


hotelName        0
url              0
zipcode          0
totalRatings     0
hotelRate        0
totalPhotos      0
priceRange       0
highlights_1    26
highlights_2    27
highlights_3    27
dtype: int64

## 1. What are the values of H2 and H3

In [60]:
df[df['highlights_2']=='none'].describe()

zipcode  totalRatings  hotelRate  totalPhotos  priceRange
count     27.000000     27.000000  27.000000    27.000000   27.000000
mean   21026.962963     14.185185   2.777778    32.111111    1.925926
std      910.548925      8.028972   1.179418    28.069054    0.474417
min    20002.000000      2.000000   0.000000     0.000000    1.000000
25%    20030.000000      8.500000   2.500000    16.000000    2.000000
50%    20770.000000     13.000000   3.000000    31.000000    2.000000
75%    22151.500000     21.000000   3.500000    40.000000    2.000000
max    22314.000000     28.000000   4.500000   150.000000    3.000000

In [61]:
df[df['highlights_2']!='none'].describe()

zipcode  totalRatings   hotelRate  totalPhotos  priceRange
count    214.000000    214.000000  214.000000   214.000000  214.000000
mean   20915.560748    112.341121    3.415888   145.873832    2.261682
std      983.252394    117.358727    0.583361   202.915852    0.518873
min    20001.000000     11.000000    0.000000    10.000000    1.000000
25%    20007.000000     44.500000    3.000000    54.250000    2.000000
50%    20745.000000     77.000000    3.500000    97.500000    2.000000
75%    22182.000000    139.000000    4.000000   169.750000    3.000000
max    22314.000000   1073.000000    4.500000  2029.000000    4.000000

In [62]:
h2 = df.groupby('highlights_2')['hotelName'].count()
h2.sort_values(ascending = False).head(10)

highlights_2
none                       27
DC                         10
metro                       7
Metro                       6
full kitchen                4
complimentary breakfast     3
free parking                3
living                      3
Gaylord                     2
mini fridge                 2
Name: hotelName, dtype: int64

In [63]:
h3 = df.groupby('highlights_3')['hotelName'].count()
h3.sort_values(ascending = False).head(10)

highlights_3
none             27
DC               11
metro             6
White House       6
DC area           3
National Mall     3
shopping          3
Carlton           2
free parking      2
housekeeping      2
Name: hotelName, dtype: int64

## What is H1 of the obs whose H2 or H3 is metro?

### Data Cleaning

In [64]:
df['highlights_1'].replace({'Metro':'metro', 'metro station':'metro','Metro station':'metro'}
                          , inplace = True)
df['highlights_2'].replace({'Metro':'metro', 'metro station':'metro','Metro station':'metro'}
                          , inplace = True)
df['highlights_3'].replace({'Metro':'metro', 'metro station':'metro','Metro station':'metro'}
                          , inplace = True)

In [65]:
df[df['highlights_1']=='metro']

hotelName  \
198                                      Americana Hotel   
667                                   Capitol Hill Hotel   
770      Clarion Collection Hotel Arlington Court Suites   
1107                                Comfort Inn Ballston   
1532             Courtyard by Marriott Arlington Rosslyn   
1637           Courtyard by Marriott Dunn Loring Fairfax   
1733       Courtyard by Marriott New Carrollton Landover   
1907   Courtyard by Marriott Washington Capitol Hill/...   
2045   Courtyard by Marriott Washington, DC/U.S. Capitol   
2103         Crowne Plaza Crystal City - Washington D.C.   
2294                            Crystal Gateway Marriott   
2530   DoubleTree by Hilton Hotel Washington DC - Cry...   
2757   Embassy Suites by Hilton Washington DC Chevy C...   
2973   Fairfield Inn & Suites by Marriott Washington,...   
3512                   Hampton Inn & Suites Falls Church   
3627        Hampton Inn & Suites Reagan National Airport   
3728      Hampton Inn Alexandria-Old Town/King St. Metro   
3762               Hampton Inn Alexandria/Pentagon South   
3839                           Hampton Inn Silver Spring   
4107                          Hilton Alexandria Old Town   
4150                                    Hilton Arlington   
4330        Hilton Garden Inn Arlington/Courthouse Plaza   
4583            Hilton Garden Inn Washington DC/Bethesda   
6070               Hyatt Place Washington DC/White House   
7302                         One Washington Circle Hotel   
7515                     Red Lion Hotel Rosslyn Iwo Jima   
7660           Renaissance Washington, DC Downtown Hotel   
7942   Residence Inn by Marriott Alexandria Old Town ...   
8003   Residence Inn by Marriott Alexandria Old Town/...   
8064    Residence Inn by Marriott Arlington Capital View   
8165   Residence Inn by Marriott Arlington Pentagon City   
8213         Residence Inn by Marriott Bethesda Downtown   
8597                        Sheraton Pentagon City Hotel   
8871   SpringHill Suites by Marriott Alexandria Old T...   
9356                                     The Embassy Inn   
9996                                    The Quincy Hotel   
10299                                      The River Inn   
10455  The Virginian Suites, an Ascend Hotel Collecti...   
10558                            The Westin Crystal City   
10951         TownePlace Suites by Marriott Falls Church   
11246                   Washington Marriott Wardman Park   
11411         West End part of the Georgetown Collection   

                                                     url  zipcode  \
198    https://www.yelp.com/biz/americana-hotel-arlin...    22202   
667    https://www.yelp.com/biz/capitol-hill-hotel-wa...    20003   
770    https://www.yelp.com/biz/clarion-collection-ho...    22201   
1107   https://www.yelp.com/biz/comfort-inn-ballston-...    22201   
1532   https://www.yelp.com/biz/courtyard-by-marriott...    22209   
1637   https://www.yelp.com/biz/courtyard-by-marriott...    22180   
1733   https://www.yelp.com/biz/courtyard-by-marriott...    20785   
1907   https://www.yelp.com/biz/courtyard-by-marriott...    20003   
2045   https://www.yelp.com/biz/courtyard-by-marriott...    20002   
2103   https://www.yelp.com/biz/crowne-plaza-crystal-...    22202   
2294   https://www.yelp.com/biz/crystal-gateway-marri...    22202   
2530   https://www.yelp.com/biz/doubletree-by-hilton-...    22202   
2757   https://www.yelp.com/biz/embassy-suites-by-hil...    20015   
2973   https://www.yelp.com/biz/fairfield-inn-and-sui...    20001   
3512   https://www.yelp.com/biz/hampton-inn-and-suite...    22042   
3627   https://www.yelp.com/biz/hampton-inn-and-suite...    22202   
3728   https://www.yelp.com/biz/hampton-inn-alexandri...    22314   
3762   https://www.yelp.com/biz/hampton-inn-alexandri...    22302   
3839   https://www.yelp.com/biz/hampton-inn-silver-sp...    20910   
4107   https://www.yelp.com/biz/hilton-alexandria-old...    22314   
4150   https://ww

In [66]:
df[(df['highlights_1']=='metro') & (df['highlights_2']=='metro')]

Empty DataFrame
Columns: [hotelName, url, zipcode, totalRatings, hotelRate, totalPhotos, priceRange, highlights_1, highlights_2, highlights_3]
Index: []

In [67]:
df[(df['highlights_2']=='metro')  &(df['highlights_3']=='metro')]

Empty DataFrame
Columns: [hotelName, url, zipcode, totalRatings, hotelRate, totalPhotos, priceRange, highlights_1, highlights_2, highlights_3]
Index: []

In [68]:
df[(df['highlights_1']=='metro') &(df['highlights_3']=='metro')]

Empty DataFrame
Columns: [hotelName, url, zipcode, totalRatings, hotelRate, totalPhotos, priceRange, highlights_1, highlights_2, highlights_3]
Index: []

> There is no duplication of metro

In [72]:
df[(df['highlights_1']=='metro') | (df['highlights_2']=='metro')
   | (df['highlights_3']=='metro')].describe()

zipcode  totalRatings  hotelRate  totalPhotos  priceRange
count     66.000000     66.000000  66.000000    66.000000   66.000000
mean   21288.363636    103.136364   3.446970   109.787879    2.166667
std     1047.149958     92.567794   0.489339    98.893340    0.414481
min    20001.000000     11.000000   2.000000    20.000000    1.000000
25%    20024.000000     46.750000   3.000000    48.250000    2.000000
50%    22150.000000     78.500000   3.500000    79.500000    2.000000
75%    22202.000000    111.750000   4.000000   126.000000    2.000000
max    22314.000000    441.000000   4.500000   594.000000    3.000000

In [70]:
df[(df['highlights_1']!='metro') & (df['highlights_2']!='metro')
   & (df['highlights_3']!='metro')].describe()

zipcode  totalRatings   hotelRate  totalPhotos  priceRange
count    175.000000    175.000000  175.000000   175.000000  175.000000
mean   20792.148571    100.668571    3.305714   141.931429    2.245714
std      911.941122    122.452589    0.765245   219.924245    0.559322
min    20001.000000      2.000000    0.000000     0.000000    1.000000
25%    20006.500000     30.000000    3.000000    43.000000    2.000000
50%    20740.000000     62.000000    3.500000    88.000000    2.000000
75%    22042.000000    137.000000    3.500000   169.500000    3.000000
max    22314.000000   1073.000000    4.500000  2029.000000    4.000000

### no significant effect either on priceRange or  on hotelRate

In [82]:
df['metro']=0

In [89]:
df.loc[(df['highlights_1']=='metro') |
            (df['highlights_2']=='metro')|
           (df['highlights_3']=='metro'),'metro']=1

## Effect of variables on price

In [100]:
import statsmodels.api as sm
model= sm.MNLogit(df['priceRange'], df[['hotelRate','totalRatings','totalPhotos','metro']])
answer= model.fit()
answer.summary()

         Current function value: 0.617131
         Iterations: 35


/opt/anaconda3/envs/python38/lib/python3.8/site-packages/statsmodels/base/model.py:567: ConvergenceWarning: Maximum Likelihood optimization failed to converge. Check mle_retvals
  warn("Maximum Likelihood optimization failed to converge. "


<class 'statsmodels.iolib.summary.Summary'>
"""
                          MNLogit Regression Results                          
==============================================================================
Dep. Variable:             priceRange   No. Observations:                  241
Model:                        MNLogit   Df Residuals:                      229
Method:                           MLE   Df Model:                            9
Date:                Wed, 03 Jun 2020   Pseudo R-squ.:                  0.1756
Time:                        16:11:17   Log-Likelihood:                -148.73
converged:                      False   LL-Null:                       -180.41
Covariance Type:            nonrobust   LLR p-value:                 2.993e-10
================================================================================
priceRange=2       coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------
hotelRate        0.1914      0.203      0.945      0.345      -0.206       0.588
totalRatings     0.0028      0.021      0.135      0.893      -0.038       0.043
totalPhotos      0.0491      0.025      1.987      0.047       0.001       0.098
metro            0.4126      1.187      0.348      0.728      -1.913       2.738
--------------------------------------------------------------------------------
priceRange=3       coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------
hotelRate       -0.5472      0.223     -2.459      0.014      -0.983      -0.111
totalRatings     0.0103      0.021      0.494      0.621      -0.030       0.051
totalPhotos      0.0542      0.025      2.182      0.029       0.006       0.103
metro           -0.1507      1.249     -0.121      0.904      -2.600       2.298
--------------------------------------------------------------------------------
priceRange=4       coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------
hotelRate       -1.0295      0.282     -3.651      0.000      -1.582      -0.477
totalRatings    -0.0028      0.023     -0.125      0.901      -0.047       0.042
totalPhotos      0.0598      0.025      2.374      0.018       0.010       0.109
metro          -26.9840   7.02e+05  -3.85e-05      1.000   -1.38e+06    1.38e+06
================================================================================
"""

### interpretation

> high Pseudo R-squ.:	0.1756

> Rate & total photo is positively correlated to the priceRange between 3-4


## Effect of variableson rate

In [101]:
import statsmodels.api as sm
model1= sm.MNLogit(df['hotelRate'], df[['priceRange','totalRatings','totalPhotos','metro']])
answer1= model1.fit()
answer1.summary()

/opt/anaconda3/envs/python38/lib/python3.8/site-packages/statsmodels/discrete/discrete_model.py:4104: SpecificationWarning: endog contains values are that not int-like. Uses string representation of value. Use integer-valued endog to suppress this warning.
  warnings.warn(msg, SpecificationWarning)
/opt/anaconda3/envs/python38/lib/python3.8/site-packages/statsmodels/base/model.py:567: ConvergenceWarning: Maximum Likelihood optimization failed to converge. Check mle_retvals
  warn("Maximum Likelihood optimization failed to converge. "


         Current function value: 1.468554
         Iterations: 35


<class 'statsmodels.iolib.summary.Summary'>
"""
                          MNLogit Regression Results                          
==============================================================================
Dep. Variable:              hotelRate   No. Observations:                  241
Model:                        MNLogit   Df Residuals:                      217
Method:                           MLE   Df Model:                           18
Date:                Wed, 03 Jun 2020   Pseudo R-squ.:                 0.06288
Time:                        16:12:04   Log-Likelihood:                -353.92
converged:                      False   LL-Null:                       -377.67
Covariance Type:            nonrobust   LLR p-value:                 0.0001789
=================================================================================
  hotelRate=2       coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
priceRange       -0.4951      0.557     -0.890      0.374      -1.586       0.596
totalRatings      0.1391      0.074      1.892      0.058      -0.005       0.283
totalPhotos      -0.0400      0.025     -1.581      0.114      -0.090       0.010
metro            24.9225   5.39e+05   4.62e-05      1.000   -1.06e+06    1.06e+06
---------------------------------------------------------------------------------
hotelRate=2.5       coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
priceRange       -0.5709      0.498     -1.147      0.251      -1.547       0.405
totalRatings      0.1384      0.073      1.903      0.057      -0.004       0.281
totalPhotos      -0.0228      0.021     -1.069      0.285      -0.065       0.019
metro            24.5359   5.39e+05   4.55e-05      1.000   -1.06e+06    1.06e+06
---------------------------------------------------------------------------------
 hotelRate=3       coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------
priceRange      -0.2683      0.471     -0.570      0.569      -1.191       0.655
totalRatings     0.1315      0.073      1.811      0.070      -0.011       0.274
totalPhotos     -0.0159      0.021     -0.761      0.447      -0.057       0.025
metro           26.2490   5.39e+05   4.87e-05      1.000   -1.06e+06    1.06e+06
--------------------------------------------------------------------------------
hotelRate=3.5       coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
priceRange        0.0794      0.465      0.171      0.865      -0.833       0.992
totalRatings      0.1288      0.073      1.774      0.076      -0.013       0.271
totalPhotos      -0.0154      0.021     -0.740      0.459      -0.056       0.025
metro            25.8224   5.39e+05   4.79e-05      1.000   -1.06e+06    1.06e+06
---------------------------------------------------------------------------------
 hotelRate=4       coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------
priceRange      -0.1789      0.470     -0.380      0.704      -1.101       0.743
totalRatings     0.1270      0.073      1.748      0.081      -0.015       0.269
totalPhotos     -0.0144      0.021     -0.688      0.491      -0.055       0.027
metro           26.1511   5.39e+05   4.85e-05      1.000   -1.06e+06    1.06e+06
--------------------------------------------------------------------------------
hotelRate=4.5       coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
priceRange       -0.3886      0.510     -0.762      0.446      -1.388       0.611
totalRatings      0.1060      0.073      1.447      0.148